### DISCUSSION TITLE
Unique Constraint

### DISCUSSION THREAD

**Discussion by TheDome0** [komapper/komapper#602] [Q&A]
There is currently no annotation to make a column unique, is there?

All I can find by searching the docs is `org.komapper.core.UniqueConstraintException`.
Seems like the only way to handle this rn is to use `onDuplicateKeyIgnore()`

**Comment by nakamura-to** [comment]
We have no annotation to make a column unique.
To make a column unique, issue an ALTER TABLE command as follows: 

```
val query = QueryDsl.executeScript("ALTER TABLE TEST ADD CONSTRAINT NAME_UNIQUE UNIQUE(NAME);")
db.runQuery(query)
```

We think DDL scripts should manage constraints.

**Reply by TheDome0** [reply]
Yes, ofc. The exception just made me wonder if there are any constraint annotations I'm missing.
